### PR TITLE
Donations block: Remove conflicting styles with themes

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/jetpack/issues/16749

#### Changes proposed in this Pull Request:

Removes the `wp-block-button_link` classes we were reusing for setting button styles for some elements in favor of fixed styles defined in the block stylesheets, making sure themes don't collide with them.

Theme | View | Before | After
--- | ---| --- | ---
Alves | Editor | Screen Shot 2020-08-26 at 11 45 23 | Screen Shot 2020-08-26 at 12 53 24
Alves | Frontend | Screen Shot 2020-08-26 at 11 46 00 | Screen Shot 2020-08-26 at 12 53 43
Twenty Twenty | Editor | Screen Shot 2020-08-26 at 12 43 28 | Screen Shot 2020-08-26 at 12 51 04
Twenty Twenty | Frontend | Screen Shot 2020-08-26 at 12 43 45 | Screen Shot 2020-08-26 at 12 51 48

#### Jetpack product discussion
pbMlHh-dW-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Set up a test site:
  * [Create a JN site running this branch](https://jurassic.ninja/create/?jetpack-beta&branch=update/donations-colors). 
  * Go to WP Admin > Settings > Jetpack Constants.
  * Set `JETPACK__SANDBOX_DOMAIN` with your sandbox address.
  * Sandbox the API and the store (see PCYsg-lW4-p2 #sandbox-method for detailed instructions).
  * Set up Jetpack and purchase a Premium or Professional plan.
* Download the Varia and Alves themes from here:
  * Varia: https://www.dropbox.com/s/iozerkwcn9wfizy/varia.zip?dl=0
  * Alves: https://www.dropbox.com/s/xyf3b285z5t0m27/alves.zip?dl=0
  * Alternatively, download them from the `master` branch of https://github.com/Automattic/themes. Do not use links available in https://wordpress.com/theme/ since they don't provide the latest version of the themes.
* Go to WP Admin > Appearance > Themes.
* Upload both of them, and activate Alves.
* Go to WP Admin > Posts > New.
* Insert a Donations block and connect to Stripe.
* Back in the editor, make sure the theme styles does not mess up the UI of the block.
* Publish the post and visit it.
* Make sure the theme styles does not mess up the UI of the block.

#### Proposed changelog entry for your changes:
Donations block: Add color settings

### CODE REVIEW

**Comment by kwight** [comment]
This works, but I find the color settings to be overwhelming – it'll be easy for users to make a mess 🙃 (and giving this many options independent of theme styles and a11y best practices feels kinda [not-WordPress](https://wordpress.org/about/philosophy/#:~:text=Decisions%2C%20not%20Options,this%20ultimately%20leads%20to%20frustration.)). I think we want any theme editor styles to come through, just without modifying the shape or layout of the donations block.

> It also removes the wp-block-button_link classes we were reusing for setting button styles for some elements in favor of fixed styles defined in the block stylesheets, making sure themes don't collide with them.

Can we just leave it at that, and drop the color settings?..

File: extensions/blocks/donations/controls.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
```
Comment (krymson24) [inline extensions/blocks/donations/controls.js:2]: @automattic is considered external? :O

Comment (jeherve) [reply extensions/blocks/donations/controls.js:2]: :) Yes, as the source does not live in this repo.

**Comment by mmtr** [comment]
> This works, but I find the color settings to be overwhelming – it'll be easy for users to make a mess 🙃 (and giving this many options independent of theme styles and a11y best practices feels kinda [not-WordPress](https://wordpress.org/about/philosophy/#:~:text=Decisions%2C%20not%20Options,this%20ultimately%20leads%20to%20frustration.)).

Yeah, I had that feeling when developing this. This would be definitely easier if we were using inner blocks, but the tabbed interface does not pair nicely with the block navigation. Blocks nested in an inactive tab would be hidden in the UI, but selectable through the block navigation:

Screen Shot 2020-08-26 at 11 15 35

> > It also removes the wp-block-button_link classes we were reusing for setting button styles for some elements in favor of fixed styles defined in the block stylesheets, making sure themes don't collide with them.
> 
> Can we just leave it at that, and drop the color settings?..

Good call! Will focus this PR on fixing only the themes styles conflicts. Let's revisit the color settings in a follow-up.

**Comment by kwight** [comment]
Works for me! <3

My only other maybe-too-far, non-blocking suggestion is to override Gutenberg's link style with a `border-radius: inherit`, for cases like this (in Twenty Seventeen):

**Comment by mmtr** [comment]
> My only other maybe-too-far, non-blocking suggestion is to override Gutenberg's link style with a `border-radius: inherit`, for cases like this (in Twenty Seventeen):

I think that's actually an issue of the theme due to not being Gutenberg ready. [Gutenberg defines a `border-radius: 20px` by default to all button blocks](https://github.com/WordPress/gutenberg/blob/44e1bfaa72688ece33aae8d50e1d921e92567d75/packages/block-library/src/button/style.scss#L9) which is of course not overridden by Twenty Seventeen because it does not support the block editor. 

Since the Donations block is intentionally using the button block markup to make the donate button to look like a button block, you'll end up with discrepancies like this when using a theme without blocks support.

**Comment by mmtr** [comment]
@Automattic/jetpack-crew quick heads up to note that this PR changes the block output, so we'd need to include these changes in Jetpack 8.9 so users can use the block safely (and without us needing to add a block deprecation).

**Review by jeherve — APPROVED** [review]
This tests well for me. Merging this now, I'll leave the WordPress.com counterpart to you.

**Comment by jeherve** [comment]
Cherry-picked to `branch-8.9` in c0145732e9e4f8f57c3b0d2505a108ef2dfe3f1a

**Comment by mmtr** [comment]
Deployed to WP.com in r212758-wpcom

**Pull request merged**